packages/shorebird_cli/test/src/cache_test.dart

```diff
@@ -198,20 +198,36 @@ void main() {
 
     group('updateAll', () {
       group('patch', () {
-        test('throws CacheUpdateFailure if a SocketException is thrown',
-            () async {
-          const exception = SocketException('test');
-          when(() => httpClient.send(any())).thenThrow(exception);
-          await expectLater(
-            runWithOverrides(cache.updateAll),
-            throwsA(
-              isA<CacheUpdateFailure>().having(
-                (e) => e.message,
-                'message',
-                contains('Failed to download patch: $exception'),
+        group('when an exception happens', () {
+          test('throws CacheUpdateFailure', () async {
+            const exception = SocketException('test');
+            when(() => httpClient.send(any())).thenThrow(exception);
+            await expectLater(
+              runWithOverrides(cache.updateAll),
+              throwsA(
+                isA<CacheUpdateFailure>().having(
+                  (e) => e.message,
+                  'message',
+                  contains('Failed to download patch: $exception'),
+                ),
               ),
-            ),
-          );
+            );
+          });
+
+          test('retries and log', () async {
+            const exception = SocketException('test');
+            when(() => httpClient.send(any())).thenThrow(exception);
```

Comment:
fwiw, our httpClient already retries. https://github.com/shorebirdtech/shorebird/blob/main/packages/shorebird_cli/lib/src/http_client/http_client.dart#L9-L14

This is probably more likely to address the case where the downloaded file's hash doesn't match what we expect